packages/mgt-chat/src/statefulClient/StatefulGraphChatListClient.ts

```diff
@@ -256,127 +205,31 @@ class StatefulGraphChatListClient implements StatefulClient<GraphChatListClient>
    */
   private notifyChatMessageEventChange(message: ChatListEvent) {
-    this._messageSubscribers.forEach(handler => handler(message));
-  }
-
-  /**
-   * Utility to convert a Graph Chat Message to an ACS Chat Message
-   * Some graph chat messages can synchronously be converted to ACS chat messages
-   * Others require an async call to get the ACS chat message
-   * Some messages can provide an initial state for immediate rendering while the async call is in progress
-   * This method returns an object containing the current value and a Promise of a future value
-   *
-   * @private
-   * @param {ChatMessage} message
-   * @return {*}  {MessageConversion}
-   * @memberof StatefulGraphChatListClient
-   */
-  private convertChatMessage(message: ChatMessage): MessageConversion {
-    switch (message.messageType) {
-      case 'message':
-        return this.graphChatMessageToAcsChatMessage(message, this.userId);
-      case 'systemEventMessage':
-      case 'unknownFutureValue':
-        return { futureValue: this.buildSystemContentMessage(message) } as MessageConversion;
-      default:
-        throw new Error(`Unknown message type ${message.messageType?.toString() || 'undefined'}`);
-    }
-  }
-
-  /**
-   * Convert an event type to the name of an icon registered in registerIcons.tsx
-   *
-   * @param eventType
-   * @returns
-   */
-  private readonly resolveIcon = (eventType: MessageEventType): string => {
-    switch (eventType) {
-      case '#microsoft.graph.membersAddedEventMessageDetail':
-        return 'add-friend';
-      case '#microsoft.graph.membersDeletedEventMessageDetail':
-        return 'left-chat';
-      case '#microsoft.graph.chatRenamedEventMessageDetail':
-        return 'edit-svg';
-      default:
-        return 'Unknown';
-    }
-  };
-
-  private async buildSystemContentMessage(message: ChatMessage): Promise<SystemMessage> {
-    const eventDetail = message.eventDetail as ChatMessageEvents;
-    let messageContent = '';
-    const awaits: Promise<IDynamicPerson>[] = [];
-    const initiatorId = eventDetail.initiator?.user?.id;
-    // we're using getUserWithPhoto here because we want to tap into the caching that mgt-person uses to cut down on graph calls
-    if (initiatorId) {
-      awaits.push(getUserWithPhoto(this.graph, initiatorId));
-      const userIds: string[] = [];
-      if (isChatMemberChangeEvent(eventDetail)) {
-        eventDetail.members?.reduce((acc, m) => {
-          if (typeof m.id === 'string') {
-            acc.push(m.id);
-          }
-          return acc;
-        }, userIds);
-      }
-      for (const id of userIds ?? []) {
-        awaits.push(getUserWithPhoto(this.graph, id));
+    this.notifyStateChange((draft: GraphChatListClient) => {
+      if (message.type === 'chatRenamed' && message.message.eventDetail) {
+        const eventDetail = message.message.eventDetail as EventMessageDetail;
+        const chatThread = draft.chatThreads.find(c => c.id === message.message.chatId);
```

Comment:
Nit: We can escape this function earlier if there's no chatThread. It's called twice.
```suggestion
        const chatThread = draft.chatThreads.find(c => c.id === message.message.chatId);
        if (!chatThread) return;
```

Reply:
All these should be refactored in their respective tasks. I added some placeholder code for testing purposes to make sure subscription events can be consumed correctly